db/migrate/20190621160800_rename_host_group_matchers.rb

```diff
@@ -0,0 +1,23 @@
+class RenameHostGroupMatchers < ActiveRecord::Migration[5.2]
+  def up
+    host_group_matchers = Setting.find_by_name('host_group_matchers_inheritance')
+    matchers_inheritance = Setting.find_by_name('matchers_inheritance')
+    return unless host_group_matchers.present? && matchers_inheritance.present?
+    matchers_inheritance.update_attribute(
+      :value,
+      host_group_matchers.value
+    )
+    host_group_matchers.destroy
+  end
+
+  def down
+    host_group_matchers = Setting.find_by_name('host_group_matchers_inheritance')
```

Comment:
I didn't get `down` method purpose. As you are destroying setting `host_group_matchers_inheritance`  in `up` method then this line always return nil. Would it be possible to use just `change` method?
or Am I missing something here?

Reply:
@kgaikwad, sorry, you are right, change is the way to go.

Reply:
@bagasse, 
Thank you for adding `change` method. 
Inside the method you mentioned `rename_column` but `host_group_matchers_inheritance` is not a column of `settings` table. 

I was expecting something like -
```
def change
  hostgroup_matcher_setting = Setting.find_by_name('host_group_matchers_inheritance')
  hostgroup_matcher_setting.update_attributes!(:name => 'matchers_inheritance')  if hostgroup_matcher_setting
end
```

Reply:
:+1: 
Yeah, sure! You can use `up and down` in this case. 
Even executing SQL query to update a setting name will also work.

Reply:
@kgaikwad, ok, after some digging, we cannot rename settings, that's why we are getting value from the old one, copy it to the new one and then deleting the old setting entry, so I rolled back to the initial implementation

Reply:
@bagasse, did you try using SQL query directly?

For example -

`execute "UPDATE settings SET name='matchers_inheritance' WHERE name='host_group_matchers_inheritance'"`

Reply:
@kgaikwad, the settings is not created via the migration but in the ` app/models/setting/puppet.rb`, the setting already exist when the migration run so renaming the old settings cannot be done because an entry with this name already exist and the name must be unique (this is enforced on the database side), so the SQL statement will not work.

Reply:
@kgaikwad, for info I've just applied the same pattern as in `db/migrate/20180228132500_rename_trusted_hosts.rb`